packages/compiler/core/types.ts

```diff
@@ -99,7 +99,7 @@ export interface Projector {
 
 export interface IntrinsicType extends BaseType {
   kind: "Intrinsic";
-  name: "ErrorType" | "void" | "never";
+  name: "ErrorType" | "void" | "never" | "unknown";
```

Comment:
I was expecting/hoping we'd get rid of the model-based intrinsics with this change. It's really easy to make bugs where you check `x.kind === "Model"` but don't do `&& !isIntrinsic(x)`.

This could maybe be left as a follow-up.

Reply:
are you talking about our Intrinsic Models ? this is for the `IntrinsicType` which is not a model.

Reply:
I'm wondering if we can make all intrinsic types go here and stop having "models" marked `@intrinsic`
name: "ErrorType" | "void" | "never" | "unknown" | "int32" | "string" | ...

Reply:
I just commented here to mean moving those out of models and into here. But this doesn't have to be done as part of this PR and should probably be discussed separately.

Reply:
the thing is all those types are `model` and if you have `model Foo is string` this also should carry over the information of `string`. I think `model` is maybe a confusing word as it kind of related to a structured object not necessary a primitive or array. Maybe a better name for it would be `type` but then you have `TypeType` in TS  which is also confusing

Reply:
`model X is string` does mess with what I wanted here.

But just generally, I find it very error-prone that you can easily get lured into treating `int32` as an object type with no properties. 

@bterlson seemed to have the same thought here: https://github.com/microsoft/cadl/pull/571#discussion_r900353588

Reply:
hhm I guess I misunderstood Brian comment there thinking it was just about `any`(now `unknown`).

Reply:
I filed https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwoKdXNpbmcgQ2FkbC5IdHRwOwoKbW9kZWwgV2lkZ2V0IGlzIHN0csQjewogIGFub3RoZXJQcm9wOscYOwp9CgpAcm91dGUoIi8iKQppbnRlcmZhY2XHQ1NlcnZpY2XFQEDEU2xpc3QoKTrHH1tdxEI%3D

with some of the weird consequences I see for allowing these to be models.

Reply:
Resolving and we can discuss on #737, especially since this PR already resolved 2/3rds of the issues I had, just differently from what I asked for here.